test/supernode_test.cpp

```diff
@@ -167,73 +166,55 @@ TEST_F(SupernodeTestSuite, TestBatchUpdate) {
     updates[i] = static_cast<vec_t>(rand() % vec_size);
   }
   auto seed = rand();
-  Supernode supernode(vec_size, seed);
-  Supernode supernode_batch(vec_size, seed);
+  Supernode::SupernodeUniquePtr supernode = Supernode::makeSupernode(vec_size, seed);
+  Supernode::SupernodeUniquePtr supernode_batch = Supernode::makeSupernode(vec_size, seed);
   auto start_time = std::chrono::steady_clock::now();
   for (const auto& update : updates) {
-    supernode.update(update);
+    supernode->update(update);
   }
   std::cout << "One by one updates took " << static_cast<std::chrono::duration<long double>>(std::chrono::steady_clock::now() - start_time).count() << std::endl;
   start_time = std::chrono::steady_clock::now();
-  Supernode* delta_node = Supernode::delta_supernode(vec_size, seed, updates);
-  supernode_batch.apply_delta_update(delta_node);
-  delete(delta_node);
+  auto delta_node = Supernode::delta_supernode(vec_size, seed, updates);
+  supernode_batch->apply_delta_update(delta_node.get());
   std::cout << "Batched updates took " << static_cast<std::chrono::duration<long double>>(std::chrono::steady_clock::now() - start_time).count() << std::endl;
 
-  ASSERT_EQ(supernode.logn, supernode_batch.logn);
-  ASSERT_EQ(supernode.idx, supernode_batch.idx);
-  for (int i=0;i<supernode.logn;++i) {
-    ASSERT_EQ(*supernode.sketches[i], *supernode_batch.sketches[i]);
+  ASSERT_EQ(supernode->logn, supernode_batch->logn);
+  ASSERT_EQ(supernode->idx, supernode_batch->idx);
+  for (int i=0;i<supernode->logn;++i) {
+    ASSERT_EQ(*supernode->get_sketch(i), *supernode_batch->get_sketch(i));
   }
 }
 
 void apply_delta_to_node(Supernode* node, const std::vector<vec_t>& updates) {
-  auto* delta_node = Supernode::delta_supernode(node->n, node->seed, updates);
-  node->apply_delta_update(delta_node);
-  delete(delta_node);
+  auto delta_node = Supernode::delta_supernode(node->n, node->seed, updates);
+  node->apply_delta_update(delta_node.get());
 }
 
-TEST_F(SupernodeTestSuite, TestConcurrency) {
```

Comment:
why did we get rid of this test?

Reply:
I have no idea. I'll try to add it back in.

Reply:
Added back in